tests/performance/test_data_analysis/statistical_data_analyzer.py

```diff
@@ -0,0 +1,274 @@
+                output += table.get_string() + "\n\n"
+
+        return output
+
+
+class StatisticalComparator:
+    """Class that contains the necessary methods to perform a comparison between the statistics
+    of a certain process and a metric. The comparison is made around a threshold value which is
+    set in the YML file.
+    """
+
+    def calculate_basic_statistics(self, dataframe: pd.DataFrame, metric: str, stat: str) -> float:
+        """Calculates the basic statistic of a data set for a specific metric.
+
+        Args:
+            dataframe (pd.DataFrame): Dataframe containing the specific data for the calculation.
+            metric (str): Metrics on which the statistic will be calculated.
+            stat (str): statistics to be calculated.
+
+        Returns:
+            value (float): value of the calculated statistic.
+        """
+        value = 0
+        if stat == 'Mean':
+            value = round(float(dataframe[metric].mean()), 2)
+        elif stat == 'Median':
+            value = round(float(dataframe[metric].median()), 2)
+        elif stat == 'Max value':
+            value = round(float(dataframe[metric].max()), 2)
+        elif stat == 'Min value':
+            value = round(float(dataframe[metric].min()), 2)
+        elif stat == 'Standard deviation':
+            value = round(float(dataframe[metric].std()), 2)
+        elif stat == 'Variance':
+            value = round(float(dataframe[metric].var()), 2)
+
+        return value
+
+    def comparison_basic_statistics(self, baseline: pd.DataFrame, datasource: pd.DataFrame, metric: str,
+                                    stat: str, threshold: float) -> int:
+        """Compares the percent change in a statistic between the two data sets to determine
+        if there is a significant change based on the threshold value.
+
+        Args:
+            baseline (pd.DataFrame): Dataframe with the baseline data.
+            datasource (pd.DataFrame): Dataframe with the incoming data source.
+            metric (str): metric from which the statistic to be compared are obtained.
+            stat (str): concrete statistic to be compared.
+            threshold (float): threshold for comparison.
+
+        Returns:
+            discrepancie (int): if the percentage difference is greater than the threshold,
+            it returns 1, otherwise it returns 0.
+        """
+        discrepancy = 0
+        baseline_value = self.calculate_basic_statistics(baseline, metric, stat)
+        dataframe_value = self.calculate_basic_statistics(datasource, metric, stat)
+
+        if baseline_value != 0:
+            diff = abs(baseline_value - dataframe_value) / baseline_value
+        else:
+            diff = abs(baseline_value - dataframe_value)
+
+        if diff >= threshold:
+            discrepancy = 1
+
+        return discrepancy
+
+
+class StatisticalTests:
+    """Class responsible for performing statistical tests on two sets of data, which allows
```

Comment:
Same here, the first line should be a summary. Detailed information should be in a description section, separated by one line

Reply:
Solved in https://github.com/wazuh/wazuh-qa/pull/5622/commits/a8be0593badbbd04bb3211468f64ad2689d17bf3